gapic-generator-java/src/main/java/com/google/api/generator/gapic/protowriter/Writer.java

```diff
@@ -69,16 +68,14 @@ public static CodeGeneratorResponse write(
       writeSamples(gapicClazz, getSamplePackage(gapicClazz), classPath, jos);
     }
 
-    writeMetadataFile(context, writePackageInfo(gapicPackageInfo, codeWriter, jos), jos);
-    writeReflectConfigFile(gapicPackageInfo.packageInfo().pakkage(), reflectConfigInfo, jos);
-
-    try {
-      jos.finish();
-      jos.flush();
-    } catch (IOException e) {
-      throw new GapicWriterException(e.getMessage(), e);
+    if (!gapicPackageInfo.isEmpty()) {
```

Comment:
Can we use `context.isEmpty()` instead of `gapicPackageInfo.isEmpty()`? So that maybe we don't have add `isEmpty()` to `GapicPackageInfo`?